python/tutorials/06-fused-attention.py

```diff
@@ -187,21 +187,21 @@ def _bwd_kernel(
         # pointer to row-wise quantities in value-like data
         D_ptrs = D + off_hz * N_CTX
         l_ptrs = L + off_hz * N_CTX
-        # initialize dv amd dk
-        dv = tl.zeros([BLOCK_M, BLOCK_DMODEL], dtype=tl.float32)
+        # initialize dk amd dv
         dk = tl.zeros([BLOCK_M, BLOCK_DMODEL], dtype=tl.float32)
+        dv = tl.zeros([BLOCK_M, BLOCK_DMODEL], dtype=tl.float32)
         # k and v stay in SRAM throughout
         k = tl.load(k_ptrs)
         v = tl.load(v_ptrs)
         # loop over rows
-        for start_m in range(lo, num_block * BLOCK_M, BLOCK_M):
+        for start_m in range(lo, num_block_q * BLOCK_M, BLOCK_M):
             offs_m_curr = start_m + offs_m
             # load q, k, v, do on-chip
             q = tl.load(q_ptrs)
             # recompute p = softmax(qk, dim=-1).T
             if CAUSAL:
                 qk = tl.dot(q, tl.trans(k), out_dtype=tl.float32)
-                qk = tl.where(offs_m_curr[:, None] >= (offs_n[None, :]), qk, float("-inf"))
+                qk = tl.where(P_SEQ + offs_m_curr[:, None] >= (offs_n[None, :]), qk, float("-inf"))
             else:
                 qk = tl.dot(q, tl.trans(k), out_dtype=tl.float32)
```

Comment:
I think, I've messed this part in previous IFU, #328 contains proper code.
Should we merge it in this IFU maybe?

Reply:
Sure. I can cherry-pick that commit over